mesonbuild/dependencies/cmake.py

```diff
@@ -652,3 +652,19 @@ def get_variable(self, *, cmake: T.Optional[str] = None, pkgconfig: T.Optional[s
         if default_value is not None:
             return default_value
         raise DependencyException(f'Could not get cmake variable and no default provided for {self!r}')
+
+
+class CMakeDependencyFactory:
+
+    def __init__(self, name: T.Optional[str] = None, modules: T.Optional[T.List[str]] = None):
```

Comment:
Seems like a better API might be `modules: T.List[str] = []`.

Reply:
Are there even any users of this API where modules isn't passed?

Reply:
One must be careful of using mutable objects as default arguments; the _defaults_ change if you modify them in the function. That doesn't seem to happen here, but it's still considered an anti-pattern to use them.